### PR TITLE
components

### PR DESCRIPTION
**Pull request by gaspard**
- [x] Add Button, Field, Card, etc
- [x] Improve theming
- [x] Improve story (theme support, defaults, better display)
- [x] Create dialog block
- [ ] `styled` tests
- [ ] `story` tests
- [ ] `dialog` tests
- [ ] `theme` tests

### CODE REVIEW

File: packages/node_modules/@lucidogen/editor/src/components/ElementTag/index.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import { connect, state } from '../../app.cerebral'
+import {
+  isCustomElement,
+  isGroupElement,
+  isStringElement,
+} from '../../lib/utils/types'
+import { CustomTag } from './CustomTag'
+import { GroupTag } from './GroupTag'
+import { StringTag } from './stringTag'
```
Comment (henri-hulski) [inline packages/node_modules/@lucidogen/editor/src/components/ElementTag/index.tsx]: Should be './StringTag'.

File: packages/node_modules/@lucidogen/editor/src/components/index.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export * from './Composition'
+export * from './Editor'
+export * from './ElementTag'
+export * from './ToolBox'
```
Comment (henri-hulski) [inline packages/node_modules/@lucidogen/editor/src/components/index.ts]: Should be './Toolbox'

File: packages/node_modules/@lucidogen/tree/package.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  "scripts": {
+    "build": "tsc -p setup/tsconfig.build.json",
+    "prepare": "npm run build",
+    "stories": "parcel serve stories/index.html",
+    "test": "jest -c setup/jest.js",
+    "test:watch": "jest -c setup/jest.js --watch --coverage false"
+  },
+  "dependencies": {
+    "@lucidogen/build": "next",
+    "@lucidogen/dragdrop": "next",
```
Comment (henri-hulski) [inline packages/node_modules/@lucidogen/tree/package.json]: `"@lucidogen/editor": "next",` is missing

File: packages/node_modules/@lucidogen/tree/src/components/Node/index.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
+// FIXME: remove
+import './style.scss'
+
+import { props as propsTag, state as stateTag } from 'Cerebral'
```
Comment (henri-hulski) [inline packages/node_modules/@lucidogen/tree/src/components/Node/index.tsx]: `from 'cerebral'` is lower case

File: packages/node_modules/@lucidogen/data/src/index.ts [hunk 5]
```diff
@@ -1,49 +1,35 @@
...
-
 import { BootBlock } from '@lucidogen/boot'
-import { DataConfig } from './helpers'
+import {
+  Block,
+  Builder,
+  CerebralModules,
+  buildSequence,
+} from '@lucidogen/build'
+import { DialogBlock } from '@lucidogen/dialog'
```
[the remark this answers is not in the document]
Comment (gaspard) [reply packages/node_modules/@lucidogen/data/src/index.ts]: They should be used in the block type definition `Block< BootBlock & DialogBlock >`.

File: packages/node_modules/@lucidogen/dragdrop/src/draggable.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+          const position = { x: e.clientX, y: e.clientY }
+          dragMove({ position })
+        }
+
+        function mouseUp() {
+          document.body.style.cursor = 'default'
+          window.removeEventListener('mousemove', mouseMove)
+          window.removeEventListener('mouseup', mouseUp)
+          dragEnd()
+        }
```
[the remark this answers is not in the document]
Comment (gaspard) [reply packages/node_modules/@lucidogen/dragdrop/src/draggable.ts]: We can remove this error: we know what we are doing with function creation in the middle of code blocks.

**Pull request merged**